src/gc/gcee.cpp

```diff
@@ -662,31 +662,37 @@ void gc_heap::fire_etw_allocation_event (size_t allocation_amount, int gen_numbe
 {
     void * typeId = nullptr;
     const WCHAR * name = nullptr;
+    
+    EX_TRY
+    {
 #ifdef FEATURE_REDHAWK
-    typeId = RedhawkGCInterface::GetLastAllocEEType();
+        typeId = RedhawkGCInterface::GetLastAllocEEType();
 #else
-    TypeHandle th = GetThread()->GetTHAllocContextObj();
-    if (th != 0)
-    {
-        InlineSString<MAX_CLASSNAME_LENGTH> strTypeName;
-        th.GetName(strTypeName);
-        typeId = th.GetMethodTable();
-        name = strTypeName.GetUnicode();
-    }
+        TypeHandle th = GetThread()->GetTHAllocContextObj();
+        if (th != 0)
+        {
+            InlineSString<MAX_CLASSNAME_LENGTH> strTypeName;
```

Comment:
This has another subtle bug that would be nice to fix while you are on it: The lifetime of InlineSString ends before the event is fired. If the type name is really long and does not fit into the stack-allocated buffer, the firing of the event can access de-allocate memory.